### PR TITLE
Add tests to the content_manager module

### LINKED ISSUE #18040: Add unit test to the `content_manager` module
| Component |
| --- |
| [content_manager](https://github.com/wazuh/wazuh/tree/dev-14153-vulndet-refactor/src/shared_modules/content_manager/src) |

## Description

Currently, the `content_manager` module lacks tests, which limits our ability to ensure its proper functionality. This issue aims to add a robust set of automated tests that cover the key functionalities of the module.

## Expected

- Define and write unit/component tests for all the main functions and methods of the module. 
- These tests should verify the expected behavior and catch any potential errors or failures.
- Identify possible improvements in the implementation.

## DoD

- [x] Unit/component tests have been written for all key functions and methods of the content_manager module.
- [x] The tests run successfully without errors and detect potential issues.

### PR DESCRIPTION
**Pull request by kevin-mcm**
|Related issue|
|---|
| Closes #18040 |

## Description

This PR aims to add tests to the following classes of the `content_manager` module
- Action
- ActionOrchestrator
- ContentManager
- ContentModuleFacade
- ContentProvider

> **Note**
> Additionally, a few changes were made to the `CMakeLists.txt` files to link the tests files to the binary of the `content_manager` module

### CODE REVIEW

**Comment by kevin-mcm** [comment]
# Coverage

## Command
```Bash
act -W .github/workflows/vulnerability-scanner-tests.yml --artifact-server-path /tmp/artifacts -j vulnerability-scanner-modules
```
## Output generated by `act` 
```Bash
[Vulnerability Scanner/vulnerability-scanner-modules] ⭐ Run Main Validate coverage
[Vulnerability Scanner/vulnerability-scanner-modules]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/4-composite-5.sh] user= workdir=
| Lines coverage is: 91.1 %
| -------------------------
| PASSED: Lines coverage is greater than 90%
| -------------------------
| Functions coverage is: 90.8 %
| ----------------------------------------------
| PASSED: Functions coverage is greater than 90%
| ----------------------------------------------
[Vulnerability Scanner/vulnerability-scanner-modules]   ✅  Success - Main Validate coverage
[Vulnerability Scanner/vulnerability-scanner-modules]   ✅  Success - Main Content manager - Coverage
```

File: src/shared_modules/content_manager/tests/component/action_test.cpp [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+    EXPECT_TRUE(std::filesystem::exists(outputFolder));
+
+    EXPECT_NO_THROW(action->startActionScheduler(interval));
+
+    std::this_thread::sleep_for(std::chrono::seconds(interval + 1));
+
+    EXPECT_NO_THROW(action->stopActionScheduler());
+
+    // This file should exist because deleteDownloadedContent is not enabled
```
Comment (sebasfalcone) [inline src/shared_modules/content_manager/tests/component/action_test.cpp:159]: +1: This is good documentation

**Review by sebasfalcone — COMMENTED** [review]
Awesome work! 

Just one minor detail

**Comment by kevin-mcm** [comment]
## Coverage before rebase
```shell
| Lines coverage is: 92.7 %
| -------------------------
| PASSED: Lines coverage is greater than 90%
| -------------------------
| Functions coverage is: 92.5 %
| ----------------------------------------------
| PASSED: Functions coverage is greater than 90%
| ----------------------------------------------
```

## Coverage after rebase
```shell
| Lines coverage is: 96.1 %
| -------------------------
| PASSED: Lines coverage is greater than 90%
| -------------------------
| Functions coverage is: 93.8 %
| ----------------------------------------------
| PASSED: Functions coverage is greater than 90%
| ----------------------------------------------
```


## Report
image

**Review by Damian-Mangold — APPROVED** [review]
LGTM !

image

**Pull request merged**